src/headers/pin.h

```diff
@@ -3,7 +3,11 @@
 
 #pragma once
 
+// SPECIAL PINS.
 #define PIN_NONE 0
+#define PIN_VIRTUAL 1  // Buttons without any hardware associated to them.
```

Comment:
While not exposed on RPi Pico pins 0 and 1 are real pins available on RP2040.
It might be better to use value well outside any addressable range.
Therefore if the value is used by mistake, it's much easier to debug as the code fails on asserts in SDK.

Or better still - don't use values to convey meaning.

E.g. use struct polymorphism and create base Button and separate implementation of IOButton and VirtualButton, and isolate HW pin use in implementation of the IOButton.

[For more info on struct polymorphism see this post.](https://stackoverflow.com/questions/8194250/polymorphism-in-c)